### PR TITLE
Fix getting hostname for HOST networking

### LINKED ISSUE #24: Service host is not available for container with HOST networking
For docker container with HOST networking, this is how docker inspect looks like

```
"Networks": {
                "host": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "52ca975025b511bfc42caf02c4777bdb9ac2570c95c429339364d80d388a2eea",
                    "EndpointID": "0c56d7ee3d9df21d762b0c99ef79b1252476e0cd2cf62a1af0ea588829f1a87e",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": ""
                }
            }
```

Current implementation parses the networks and uses empty string from gateway as hostname. However I think that in that case it should use localhost instead.

### PR DESCRIPTION
**Pull request by alenkacz**
For host networked container it uses localhost as a default

fixes #24

### CODE REVIEW

File: src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy [hunk 1]
```diff
@@ -111,7 +111,9 @@ class ComposeUp extends DefaultTask {
             String gateway
             Map<String, Object> networkSettings = inspection.NetworkSettings
             Map<String, Object> networks = networkSettings.Networks
-            if (networks) {
+            if (networks && networks.every { it.key.toLowerCase().equals("host") }) {
+                gateway = 'localhost'
```
Comment (augi) [inline src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy]: Please add `logger.debug()` here.

Comment (alenkacz) [inline src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy]: done

File: src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy [hunk 2]
```diff
@@ -111,7 +111,10 @@ class ComposeUp extends DefaultTask {
             String gateway
             Map<String, Object> networkSettings = inspection.NetworkSettings
             Map<String, Object> networks = networkSettings.Networks
-            if (networks) {
+            if (networks && networks.every { it.key.toLowerCase().equals("host") }) {
+                gateway = 'localhost'
+                logger.debug("Will use $gateway as host of $serviceName beacuse it is using HOST network")
```
Comment (augi) [inline src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy]: Typo in `beacuse ` :-)

Comment (alenkacz) [inline src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy]: Typo fixed :)

**Comment by augi** [comment]
Released as 0.3.4

**Comment by pzbitskiy** [comment]
This is actually breaks tests on Windows then boot2docker VM is used:
```
test.environment.get('WEB_HOST') == 'localhost'
|    |           |               |
|    |           172.17.1.100    false
|    |                           12 differences (0% similarity)
|    |                           (172.17.1.100)
|    |                           (localhost---)
```

**Comment by alenkacz** [comment]
@pzbitskiy ha! thanks, I'll fix that...

**Pull request merged**